core/unit_test/TestQuadPrecisionMath.hpp

```diff
@@ -98,12 +100,48 @@ TEST(TEST_CATEGORY, quad_precision_common_math_functions) {
   Kokkos::parallel_for(
       Kokkos::RangePolicy<Kokkos::DefaultHostExecutionSpace>(0, 1),
       KOKKOS_LAMBDA(int) {
-        (void)Kokkos::Experimental::fabs((__float128)0);
-        (void)Kokkos::Experimental::sqrt((__float128)1);
-        (void)Kokkos::Experimental::exp((__float128)2);
-        (void)Kokkos::Experimental::sin((__float128)3);
-        (void)Kokkos::Experimental::cosh((__float128)4);
+        (void)Kokkos::fabs((__float128)0);
+        (void)Kokkos::sqrt((__float128)1);
+        (void)Kokkos::exp((__float128)2);
+        (void)Kokkos::sin((__float128)3);
+        (void)Kokkos::cosh((__float128)4);
       });
 }
 
+#define STATIC_ASSERT(...) static_assert(__VA_ARGS__, "")  // FIXME C++17
+
+constexpr bool test_quad_precision_promotion_traits() {
+  STATIC_ASSERT(
+      std::is_same<__float128, decltype(Kokkos::pow(__float128(1), 2))>::value);
+  STATIC_ASSERT(std::is_same<__float128,
+                             decltype(Kokkos::hypot(3, __float128(4)))>::value);
+  return true;
+}
+
+STATIC_ASSERT(test_quad_precision_promotion_traits());
+
+constexpr bool test_quad_precision_math_constants() {
+  // compare to mathematical constants defined in libquadmath when available
+  // clang-format off
+  STATIC_ASSERT(Kokkos::Experimental::e_v     <__float128> == M_Eq);
+  STATIC_ASSERT(Kokkos::Experimental::log2e_v <__float128> == M_LOG2Eq);
+  STATIC_ASSERT(Kokkos::Experimental::log10e_v<__float128> == M_LOG10Eq);
+  STATIC_ASSERT(Kokkos::Experimental::pi_v    <__float128> == M_PIq);
+  // STATIC_ASSERT(Kokkos::Experimental::inv_pi_v<__float128> == M_1_PIq);  // FIXME
```

Comment:
```
Static_assert failed due to requirement 'Kokkos::Experimental::inv_pi_v<__float128> == 0.318309886183790671537767526745028689Q' ""

Reply:
works with gcc >= 8.4. Do we want to include this?